src/imperative/imperative.cc

```diff
@@ -273,40 +275,34 @@ void Imperative::RecordOp(
       info.outputs.back().dtype_ = outputs[i]->dtype();
       info.outputs.back().storage_type_ = outputs[i]->storage_type();
     }
-    outputs[i]->entry_ = nnvm::NodeEntry{node, i, 0};
+    outputs[i]->entry_ = nnvm::NodeEntry{node, static_cast<uint32_t>(i), 0};
   }
 }
 
-std::vector<NDArray*> Imperative::Backward(
-    const std::vector<NDArray*>& outputs,
-    const std::vector<NDArray*>& ograds,
-    const std::vector<NDArray*>& variables,
-    bool is_train, bool retain_graph,
-    bool create_graph) {
-  using namespace nnvm;
-  using namespace imperative;
-  static const std::vector<const Op*> zero_ops{Op::Get("zeros_like"), Op::Get("_zeros")};
-  static const Op* copy_op = Op::Get("_copy");
-
-  // Construct forward graph
-  Graph graph;
-  graph.outputs.reserve(outputs.size());
-  for (const auto& i : outputs) {
+std::vector<nnvm::NodeEntry> Imperative::CreateForwardGraph(const std::vector<NDArray*>& outputs) {
```

Comment:
I found this function and naming quite confusing. It is not really creating a graph, but simply assign a vector of outputs to the graph outputs. Maybe consider creating and returning the graph itself from this function?

Reply:
Done